fe/fe-core/src/main/java/com/starrocks/sql/analyzer/Analyzer.java

```diff
@@ -9,93 +9,106 @@
-            QueryAnalyzerV2 analyzerV2 = new QueryAnalyzerV2(catalog, session);
-            analyzerV2.analyze(node);
+        @Override
+        public Void visitAnalyzeStatement(AnalyzeStmt statement, ConnectContext session) {
+            analyzeAnalyzeStmt(statement, session);
+            return null;
+        }
 
-            QueryRelation queryRelation = ((QueryStatement) node).getQueryRelation();
-            long selectLimit = ConnectContext.get().getSessionVariable().getSqlSelectLimit();
-            if (!queryRelation.hasLimit() && selectLimit != SessionVariable.DEFAULT_SELECT_LIMIT) {
-                queryRelation.setLimit(new LimitElement(selectLimit));
-            }
+        @Override
+        public Void visitBaseViewStatement(BaseViewStmt statement, ConnectContext session) {
+            ViewAnalyzer.analyze(statement, session);
+            return null;
+        }
 
-            return ((QueryStatement) node).getQueryRelation();
-        } else if (node instanceof BaseViewStmt) {
-            new ViewAnalyzer(session).analyze((BaseViewStmt) node);
+        @Override
+        public Void visitCreateAnalyzeJobStatement(CreateAnalyzeJobStmt statement, ConnectContext session) {
+            analyzeCreateAnalyzeStmt(statement, session);
             return null;
-        } else if (node instanceof InsertStmt) {
-            return new InsertAnalyzer(catalog, session).transformInsertStmt((InsertStmt) node);
-        } else if (node instanceof AnalyzeStmt) {
-            return analyzeAnalyzeStmt((AnalyzeStmt) node);
-        } else if (node instanceof CreateAnalyzeJobStmt) {
-            return analyzeCreateAnalyzeStmt((CreateAnalyzeJobStmt) node);
-        } else if (node instanceof CreateTableAsSelectStmt) {
+        }
+
+        @Override
+        public Void visitCreateTableAsSelectStatement(CreateTableAsSelectStmt statement, ConnectContext session) {
             // this phrase do not analyze insertStmt, insertStmt will analyze in
             // StmtExecutor.handleCreateTableAsSelectStmt because planner will not do meta operations
-            return new CTASAnalyzer(catalog, session).transformCTASStmt((CreateTableAsSelectStmt) node);
-        } else if (node instanceof CreateWorkGroupStmt) {
-            return ((CreateWorkGroupStmt) node).analyze();
-        } else if (node instanceof AlterWorkGroupStmt) {
-            return ((AlterWorkGroupStmt) node).analyze();
-        } else if (node instanceof DropWorkGroupStmt) {
-            return ((DropWorkGroupStmt) node).analyze();
-        } else if (node instanceof ShowStmt) {
-            new ShowStmtAnalyzer(session).analyze((ShowStmt) node);
+            CTASAnalyzer.transformCTASStmt((CreateTableAsSelectStmt) statement, session);
+            return null;
+        }
+
+        @Override
+        public Void visitCreateWorkGroupStatement(CreateWorkGroupStmt statement, ConnectContext session) {
+            statement.analyze();
```

Comment:
sometimes we need use a analyzer (like InsertAnalyzer), sometimes just use statement.analyze, have any plan to unify this ?

Reply:
Yes, it should be unified, but it will not be changed for the time being. This involves too many functions